src/spac/visualization.py

```diff
@@ -482,14 +465,17 @@ def hierarchical_heatmap(
     import matplotlib.pyplot as plt
 
     # Convert the linkage data to type double
-    dendro_row_data = np.array(dendrogram_data['row_linkage'], dtype=np.double)
-
-    # Plot the dendrogram
-    fig, ax = plt.subplots(figsize=(10, 7))
-    sch.dendrogram(dendro_row_data, ax=ax)
-    plt.title('Row Dendrogram')
-    plt.show()
-
+    dendro_col_data = np.array(dendrogram_data['col_linkage'], dtype=np.double)
+
+    # Ensure the linkage matrix has at least two dimensions and
+    more than one linkage
+    if dendro_col_data.ndim == 2 and dendro_col_data.shape[0] > 1:
+        fig, ax = plt.subplots(figsize=(10, 7))
+        sch.dendrogram(dendro_col_data, ax=ax)
+        plt.title('Standalone Col Dendrogram')
+        plt.show()

```

Comment:
@fangliu117, SPAC functions should not make plots, but return the fig, and ax to the caller.

Reply:
Thanks @georgezakinih . Sorry to make you confused. Let me clarify it.
The purpose of these lines is to provide an example of how to call the function and display the dendrogram. They don't belong to function code.